### ISSUE TITLE
Output discrepancies for detectron2 model converted to caffe2

### ISSUE THREAD

**Issue by ydas** [facebookresearch/detectron2#2761]
## Instructions To Reproduce the Issue and Full Logs

I trained a detectron2 model "model_final.pth" on my home PC and uploaded it to an AWS detectron2 docker server pulled from your github account. On that detectron2 server and using the deployment scripts, I converted the "model_final.pth" to a caffe2 model (model.pb and model_init.pb).

I can run boundary detection inferences on images on that docker server 
1) using a python script on "model_final.pth" or 
2) using a c++ program (similar to caffe2_mask_rcnn.cpp) on the caffe2 model "model.pb"

The output of the detected boundaries coming out of the converted caffe2 model  are not identical to the output from "model_final.pth". Many boundaries are identical but there are serious discrepancies. See attached example of an image giving different ouput. As both the python script and c++ program are compiled and linked to the same libs (protobuf, ...) on the same server, I assume that the problem is caused by the conversion process.

![Uploading PMC64853_003.png…]()
![PMC320490_006_b xml_b](https://user-images.githubusercontent.com/2511338/111496936-9d2fb580-8738-11eb-8b00-2c67e9eb3438.png)
![PMC320490_006_b xml](https://user-images.githubusercontent.com/2511338/111496940-9dc84c00-8738-11eb-9523-5d21297aee22.png)

## Question
Is it normal to observe such deviations between "model_final.pth" and a converted model to caffe2?
What could be the likely causes of such deviation?
Thank you

## Environment:

Failed to load OpenCL runtime
No CUDA runtime is found, using CUDA_HOME='/usr/local/cuda'
---------------------  -------------------------------------------------------------------------
sys.platform           linux
Python                 3.6.9 (default, Jan 26 2021, 15:33:00) [GCC 8.4.0]
numpy                  1.19.5
detectron2             0.4 @/home/appuser/detectron2_repo/detectron2
Compiler               GCC 7.5
CUDA compiler          CUDA 10.1
DETECTRON2_ENV_MODULE  <not set>
PyTorch                1.8.0+cu101 @/home/appuser/.local/lib/python3.6/site-packages/torch
PyTorch debug build    False
GPU available          False
Pillow                 8.1.2
torchvision            0.9.0+cu101 @/home/appuser/.local/lib/python3.6/site-packages/torchvision
fvcore                 0.1.3
cv2                    3.2.0
---------------------  -------------------------------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.7.0 (Git Hash 7aed236906b1f7a05c0917e5257a1af05e9ff683)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=10.1, CUDNN_VERSION=7.6.3, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.8.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON,

Thank you

**Comment by ydas** [comment]
Sorry. Found the non-issue : image padding error in my code (added to comply with test "assert(height % 32 == 0 && width % 32 == 0)").
